homeassistant/components/switch/shelly.py

```diff
@@ -0,0 +1,121 @@
+"""
+Support for The Shelly Wifi switch.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.shelly/
+"""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_PATH,
+    CONF_USERNAME, CONF_PASSWORD, CONF_SWITCHES)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PATH = "/relay/0"
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_PATH, default=DEFAULT_PATH): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+})
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Set up Shelly Wifi switches."""
+    switches = config.get('switches', {})
+    devices = []
+
+    for dev_name, properties in switches.items():
+        devices.append(
+            ShellySwitch(
+                hass,
+                properties.get(CONF_NAME, dev_name),
+                properties.get(CONF_HOST, None),
+                properties.get(CONF_PATH, DEFAULT_PATH),
+                properties.get(CONF_USERNAME, None),
+                properties.get(CONF_PASSWORD)))
+
+    add_devices_callback(devices)
+
+
+class ShellySwitch(SwitchDevice):
+    """Representation of a Shelly Wifi switch."""
+
+    def __init__(self, hass, name, host, path, user, passwd):
+        """Initialize the device."""
+        self._hass = hass
+        self._name = name
+        self._state = False
+        self._url = 'http://{}{}'.format(host, path)
+        if user is not None:
+            self._auth = (user, passwd)
+        else:
+            self._auth = None
+
+    def _switch(self, newstate):
+        """Switch on or off."""
+        _LOGGER.info("Switching to state: %s", newstate)
+
+        try:
+            req = requests.get('{}?turn={}'.format(self._url, newstate),
+                               auth=self._auth, timeout=5)
+            result = req.status_code
+            if result == 200:
+                return True
+            else:
+                return False
+        except requests.RequestException as error:
+            _LOGGER.error("Switching failed: " + error)
+
+    def _query_state(self):
+        """Query switch state."""
+        _LOGGER.info("Querying state from: %s", self._url)
```

Comment:
This might spam the logs a bit too much for the average user. I'm wondering if this should be `DEBUG` (since the only time you'd feasibly want the info is when debugging a connection) – thoughts?

Reply:
You are right. Polling let's say every 5 seconds can cause log spam. Moved that to debug.
Resolved here:
https://github.com/ispiropoulos/home-assistant/commit/bea446b91781a425888339ab7f7999abcfa8873a
and here:
https://github.com/ispiropoulos/home-assistant/commit/abdb78ab42db528864cdec3e9620beb12ed4d520